File: jupyter2/ps/how_compiled/tail_recursion/go/tail_recursion.go
package how_compiled

func SumArrayTail(a []float64, i int64, n int64, s float64) float64 {
	if i == n {
		return s
	} else {
		return SumArrayTail(a, i + 1, n, s + a[i])
	}
}

File: jupyter2/answers/how_compiled/tail_recursion/jl/tail_recursion.jl
function sum_array_tail(a :: Vector{Float64}, i :: Int64, n :: Int64, s :: Float64) :: Float64
    if i > n
        return s
    else
        return sum_array_tail(a, i + 1, n, s + a[i])
    end
end

import InteractiveUtils
InteractiveUtils.code_native(sum_array_tail)
